docs/proposals/portable-service-assets.md

```diff
@@ -0,0 +1,84 @@
+# Portable Service Assets
+
+## Introduction
+Service classes in the Service Catalog require a number of UI graphical assets (i.e. service icon, company logo, etc).  For APBs that are sourced from the RHCC, we do not want to reference these externally on a remote (non-RH managed endpoint).  Additionally, we want to be able to handle off-network / disconnected clusters thus making this a truly portable solution.
+
+## Problem Description
+Allow APBs to specify static file assets which will be accessible to any computer using OpenShift.
+* How do we build and store these assets for each APB?
+* How do we download these assets before downloading the entire APB since they will be needed before deploying?
+* How do we serve these assets?
+
+## Building and Storing Assets
+Assets will be stored under the APB folder in a folder named `assets`. As part of the `apb build` process a container image will be created copying all files from the `assets` folder into the container.
+
+```Dockerfile
+# Assets Dockerfile
+FROM scratch
+COPY assets /assets
+CMD ["noop"]
+```
+
+This docker container will be built with the same image name, but the tag will be prefixed with `_assets_`.  APB tooling will perform `docker build -t hello-world-apb:_assets_latest .`  and keep track of the image ID.
+
+After building the assets image, the tooling will also add a label to the APB image with the assets image ID.  This will ensure that if the APB is updated,  the matching assets will be associated and we can detect changes in the assets even if the APB spec has not changed.
+
+```bash
+docker build . -t docker.io/cfchase/hello-world-apb:latest --label "com.redhat.apb.assetsImageId=<ImageId>"
```

Comment:
I can use this method to eliminate the `LABEL "com.redhat.apb.spec"` from the Dockerfile as well and just have it as part of the build process so it's not checked in as changes.  That way our apb.yml and Dockerfile would not be out of sync anymore.  Let me know if that would be desired or if there are any reasons we want to keep it in the Dockerfile.